### ISSUE TITLE
units other than "px" ?

### ISSUE THREAD

**Issue by yairEO** [smooth-code/xstyled#36]
## 💬 Questions and Help

From the example of the [home page](https://www.smooth-code.com/open-source/xstyled/) it is unclean which unit types are used for things like padding, margin, & font-size (which isn't shown in the demo)

While the [theme page](https://www.smooth-code.com/open-source/xstyled/docs/theming/) provides a very shallow explanation to what's shown on the home page (I think it should explicitly write on the home page which units are used) the theme page does not explain how to use other unit types, such as `em` or `rem`. 

For margins or paddings, should this:

    space: [0, 1, 2]

become this?

    space: [0, "1em", "1em"]

Thanks!

**Comment by dapetrov** [comment]
Inside space array you specify indices from 0 to n (desired):

```js
  space: [0, 2. 4. 8. 12, 24]
```
 and it becomes your theme constraints, so if your use xstyled and types:

```js
.header {
  padding: 2
}
```

xstyled deals with this value as with index of the space array, so 2 becomes space[2] -> 4, and css-in-js libriaries (styled-components or emotion) convert numbers to pixels, so it becomes '4px'

if you specify this array like

```js
  space: [0, '2em', '3em', '4em']
```

2 becomes space[2] -> '3em', and css-in-js libriaries (styled-components or emotion) don't convert it, as it is a string, so it will be '3em'

**Comment by dapetrov** [comment]
So, if you use a raw value like '2' or '4' as the value of the css property xstyled deals with it like with index according to which theme group the specific css propery is related.

**Comment by yairEO** [comment]
maybe it will be nicer if you set the units of the Array outside of it, in your theme, like so:

    space: [0, 2, 4, 8, 12, 24]
    space.units = "rem"

And your wrapper would use the `units` value, if specified, else default to `px`

**Comment by neoziro** [comment]
Hello all, this is solved by [transformers](https://www.smooth-code.com/open-source/xstyled/docs/transformers/), it is available in v1.8.0.
